src/infer_request.cc

```diff
@@ -1272,20 +1274,31 @@ InferenceRequest::ReportStatistics(
     }
   } else {
     model_raw_->MutableStatsAggregator()->UpdateFailure(
-        metric_reporter, request_start_ns_, request_end_ns);
+        metric_reporter, request_start_ns_, request_end_ns, reason);
     if (secondary_stats_aggregator_ != nullptr) {
       secondary_stats_aggregator_->UpdateFailure(
-          nullptr /* metric_reporter */, request_start_ns_, request_end_ns);
+          nullptr /* metric_reporter */, request_start_ns_, request_end_ns, reason);
     }
   }
 }
 
+void 
+InferenceRequest::ReportStatistics(
+  MetricModelReporter* metric_reporter, bool success,
+  const uint64_t compute_start_ns, const uint64_t compute_input_end_ns,
+  const uint64_t compute_output_start_ns, const uint64_t compute_end_ns) {
+  // Call the updated ReportStatistics with a default error_type
+  ReportStatistics(metric_reporter, success, compute_start_ns, compute_input_end_ns, 
+                         compute_output_start_ns, compute_end_ns, FailureReason::BACKEND);
+}
```

Comment:
I think this overload is handled by your change to `TRITONBACKEND_ModelInstanceReportStatistics` to always pass `FailureReason::BACKEND` - is this one needed?

Reply:
Correct, removed the overload.